Rubberduck.CodeAnalysis/Inspections/Concrete/FunctionReturnValueNotUsedInspection.cs

```diff
@@ -111,6 +119,19 @@ private bool IsCallStmt(IdentifierReference usage)
             {
                 return false;
             }
+
+            var indexExpr = usage.Context.GetAncestor<VBAParser.IndexExprContext>();
+            if (indexExpr != null)
+            {
+                var memberAccessStmt = usage.Context.GetAncestor<VBAParser.MemberAccessExprContext>();
+                if (memberAccessStmt != null &&
+                    callStmt.SourceInterval.ProperlyContains(memberAccessStmt.SourceInterval) &&
+                    memberAccessStmt.SourceInterval.ProperlyContains(indexExpr.SourceInterval))
+                {
+                    return false;
+                }
+            }
```

Comment:
Only tangentially related to your fix, but another 2 calls to `GetAncestor<T>` makes an inefficient sieve in `IsReturnValueUsed` even more inefficient.  The predicates in the O(MG) `IsReturnValueUsed` method are in a horrible order. They should at very least be ordered in a way that short circuits on the most common usage to the least common. I'd suggest this instead:

            return (from usage in function.References
                    where !IsLet(usage)
                    where !IsSet(usage)
                    where !IsCallStmt(usage)
                    where !IsTypeOfExpression(usage)
                    where !IsAddressOfCall(usage)
                    select usage).Any(usage => !IsReturnStatement(function, usage));

Ideally, this whole thing should be re-written so that it only walks the parse tree backward **_once_** instead of potentially doing that 7 times, but that's probably out of scope for the PR.